### ISSUE TITLE
./goland does not work from (X)ubuntu

### ISSUE THREAD

**Issue by jglick** [jenkins-x/dev-env#15]
**Describe the bug**
There is some sort of problem with the X11 display variable on Ubuntu.

```
…
Status: Downloaded newer image for rycus86/goland:latest
docker: Error response from daemon: create .0: ".0" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
```

**To Reproduce**
1. Run (X)ubuntu.
2. Follow [instructions](https://jenkins-x.github.io/dev-env/docs/goland.html)

**Expected behavior**
Some IDE opens?

**Desktop (please complete the following information):**
 - OS: Xubuntu Cosmic
 - Browser N/A
 - Version N/A

**Comment by cagiti** [comment]
Probably need to detect the OS and apply the DISPLAY VARIABLE appropriately. Incidentally, what's the value of your DISPLAY variable? Thanks

**Comment by jglick** [comment]
`:0.0`

**Comment by jglick** [comment]
Something to do with

```bash
--volume ${DISPLAY/:0}:/tmp/.X11-unix
```

clearly, though I am struggling to figure out why you are using `$DISPLAY` in a volume name to begin with. The [official instructions](https://hub.docker.com/r/rycus86/goland/) suggest something that makes more sense, and in fact the following patch fixes X11 for me:

```diff
diff --git a/goland b/goland
index 765155c..aaadafb 100755
--- a/goland
+++ b/goland
@@ -19,11 +19,6 @@ function create_dirs() {
   create_dir ~/.GoLand.java
 }
 
-function setup_env_vars() {
-  PORT=$(ps -ef | grep "Xquartz :\d" | grep -v xinit | awk '{ print $9; }')
-  display="--env DISPLAY=$(hostname)${PORT}"
-}
-
 function setup_xquartz() {
   if [ -x $XHOST ]
   then
@@ -44,9 +39,9 @@ function pull_latest() {
 function run() {
   docker run --rm \
              --detach \
-             ${display} \
+             --env DISPLAY=$DISPLAY \
              --security-opt=seccomp:unconfined \
-             --volume ${DISPLAY/:0}:/tmp/.X11-unix \
+             --volume /tmp/.X11-unix:/tmp/.X11-unix \
              --volume ~/.GoLand:/home/developer/.GoLand \
              --volume ~/.GoLand.java:/home/developer/.java \
              --volume $GO_WORKSPACE:/home/developer/go \
@@ -56,6 +51,5 @@ function run() {
 
 create_dirs
 setup_xquartz
-setup_env_vars
 pull_latest
 run
```

I do not have a Mac to understand why the extra complexity is needed in that case, if it is.

**Comment by cagiti** [comment]
Yeah, I tend to agree... It does look a bit odd but it's mounting the DISPLAY socket into the container. Thanks for spending the time to look for a solution 👍

**Comment by owennewo** [comment]
@jglick I get the same errors about invalid characters when on kubuntu 18.10.

What works for me is to run the following:
> xhost +

The above command basically disables access control for connecting to x, so you may want to do something more secure!!
After the above I run

> docker run --rm   -e DISPLAY=${DISPLAY}   -v /tmp/.X11-unix:/tmp/.X11-unix   -v ~/.GoLand:/home/developer/.GoLand   -v ~/.GoLand.java:/home/developer/.java   -v ~/Project/go:/home/developer/go   --security-opt=seccomp:unconfined   --name goland-$(head -c 4 /dev/urandom | xxd -p)-$(date +'%Y%m%d-%H%M%S') rycus86/goland:latest

I'm actually wanting to use dev-env for java development so not really interested in goland.  It starts up though.  Rumour has it that @edwardmlyte might be helping out with java version?

**Comment by edwardmlyte** [comment]
Correct. In my fork on the Java branch there's intellij running in nearly
an identical manner. Having chatted with Cai the might be too much
maintenance to try and have multiple language branches that easily merge.
So I'll go off on a tangent and try to keep it as similar as possible,
whilst keeping up with Cai with new developments

**Comment by cagiti** [comment]
Hey @jglick, I've now removed the script to execute goland and moved it into a doc available from the README... its available in #27 if you're interested. Thanks 👍
